### PR TITLE
Feature views

### PR DESCRIPTION
**Pull request by unorjikingsley**
## The following requirement was executed in this PR;

- Implement the design from the sneak peek wireframes.
    - Styling is nice to have but focus on making sure everything functions first. (Note: Ignore login logic, you would be using Devise in coming milestone)

**Note: Analyze the wireframes deeply and try to identify all the necessary elements and how a user would navigate from one page to another**
    
blog_root_users 1 blog_user_page 2 blog_user_all_posts 3 blog_single_post 4

- Use methods that you have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).
- **Hint:** In order to see some posts or comments, you might need to add some data to your database as described [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md#use-models-to-insert-data).

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -3,4 +3,7 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
```
Comment (franclobo) [inline config/routes.rb:5]: - [ ] Please 🙏🏽 define the root route user#index to start with the users page when your app opens for a better UX.

File: app/views/users/index.html.erb
Comment (franclobo) [inline app/views/users/index.html.erb:1]: - [ ] Kindly add the correct styles to this page to be similar to the wireframe 👇🏽 


| Wireframe | Your design |
|---|---|
| image | image |

**Note** Whether you want to create many users, posts, and comments you could install the gem 'faker' in your Gemfile and create them in the seeds.rb file, like this 👇🏽

```ruby
require 'faker'

10.times do
  User.create(
    name: Faker::Name.name,
    photo: Faker::Avatar.image,
    bio: Faker::Lorem.paragraph(sentence_count: 2),
    posts_counter: 0
  )
end

50.times do
  Post.create(
    title: Faker::Lorem.sentence(word_count: 3),
    text: Faker::Lorem.paragraph(sentence_count: 2),
    author_id: User.all.sample.id,
    likes_counter: 0,
    comments_counter: 0
  )
end

100.times do
  Comment.create(
    text: Faker::Lorem.paragraph(sentence_count: 2),
    author_id: User.all.sample.id,
    post_id: Post.all.sample.id
  )
end
```
After that run in your terminal `rails db:drop db:create db:migrate db:seed`
Finally, run `rails s` to open it in your browser.

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<h1>Users</h1>
+<ul>
+  <% @users.each do |user| %>
+    <a href="<%= user_posts_path(user) %>">
```
Comment (franclobo) [inline app/views/users/index.html.erb]: - [ ] When I clicked on the user's name it returned the post#index page, please correct the path to go to the correct user#show page. It should open the user#show page like this wireframe 👇🏽 

image

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @unorjikingsley, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️
- Correct Git flow ✔️
- Use preprocessed HTML file with embedded Ruby code. ✔️
- Use layouts and templates for shared content. ✔️


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/index.html.erb
Comment (unorjikingsley) [reply app/views/users/index.html.erb:1]: @franclobo i installed faker but i am really struggling to resize the profile image. Any ideas on how to do that?

**Review by OyePriscilla — APPROVED** [review]
Hi @unorjikingsley 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
Implemented root path and show#page ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**